includes/Pages/RequestAction/PageManuallyConfirm.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/******************************************************************************
+ * Wikipedia Account Creation Assistance tool                                 *
+ *                                                                            *
+ * All code in this file is released into the public domain by the ACC        *
+ * Development Team. Please see team.json for a list of contributors.         *
+ ******************************************************************************/
+
+namespace Waca\Pages\RequestAction;
+
+use Waca\Exceptions\ApplicationLogicException;
+use Waca\Exceptions\OptimisticLockFailedException;
+use Waca\Helpers\Logger;
+
+class PageManuallyConfirm extends RequestActionBase
+{
+    /**
+     * This endpoint manually confirms a request, bypassing email confirmation.
+     *
+     * Only administrators are allowed to do this, for obvious reasons.
+     *
+     * @throws ApplicationLogicException|OptimisticLockFailedException
+     */
+    protected function main()
+    {
+        // This method throws an error if we don't post
+        $this->checkPosted();
+
+        // Retrieve the database.
+        $database = $this->getDatabase();
+
+        // Find the request
+        // This method throws exceptions if there is an error with the request.
+        $request = $this->getRequest($database);
+
+        // Mark the request as confirmed.
+        $request->setEmailConfirm("Confirmed");
+        $request->save();
```

Comment:
We should pass through the updateVersion too, along with the request ID. This gives us [a form of locking](https://en.wikipedia.org/wiki/Optimistic_concurrency_control), by telling the tool "the user last saw version X of this record" (which you can pass in via `$request->setUpdateVersion(...)`). If the version of the row has changed, it means there's an edit conflict, and we should get the user to browse back and check they still want to take this action. Comparing the value passed in against the value in the database and generating the error is done automatically for you by `->save()`, but you still need to tell it the updateVersion the user last saw.

PageFlagComment does this if you want an example, or let me know if you want me to walk you through the changes needed.